test/hash_recursive_variant_test.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2011-2014
+// Antony Polukhin
+//
+// Distributed under the Boost Software License, Version 1.0. (See
+// accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#include <unordered_set>
```

Comment:
Use macro from Boost.Config or just make sure that compiler is c++11 and disable the tests if it is not.